src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -180,6 +190,32 @@ public Task StopAsync()
             return Task.CompletedTask;
         }
 
+        public async Task CloseAsync()
+        {
+            if (!_closeConnectionLock.Wait(0))
+            {
+                // only one close async process can exist at a time.
+                return;
+            }
+
+            try
+            {
+                Console.WriteLine($"{HubEndpoint}: sending fin request.");
+                await WriteAsync(new CloseConnectionMessage(Constants.Fin));
+                await _incomingTask;
+                Console.WriteLine($"{HubEndpoint}: fin ack received.");
+            } catch (Exception)
+            {
+                // simply skip the awaiting incoming task step if any error occured.

```

Comment:
Log the exception as debug.

Reply:
I Logged this as a warning, similar to `Start` and `Stop`.